### PR TITLE
feature/order-add-payment-terms-to-order

### PR DESCRIPTION
**Pull request by DamienDelprat**
- add payment terms to order.go

[Doc](https://shopify.dev/docs/api/admin-rest/2024-04/resources/order#resource-object) :
image

### CODE REVIEW

**Comment by DamienDelprat** [comment]
Where does the merge conflict come from ?

**Comment by oliver006** [comment]
> Where does the merge conflict come from ?

No idea, try rebasing?

File: order.go [hunk 1]
```diff
@@ -285,6 +286,24 @@ type RefundLineItem struct {
 	TotalTax   *decimal.Decimal `json:"total_tax,omitempty"`
 }
 
+type PaymentTerms struct {
+	Amount           *decimal.Decimal  `json:"amount,omitempty"`
+	Currency         string            `json:"currency,omitempty"`
+	PaymentTermsName string            `json:"payment_terms_name,omitempty"`
+	PaymentTermsType string            `json:"payment_terms_type,omitempty"`
+	DueInDays        string            `json:"due_in_days,omitempty"`
```
Comment (oliver006) [inline order.go:294]: This should be an int?

File: order.go [hunk 2]
```diff
@@ -285,6 +286,24 @@ type RefundLineItem struct {
...
+	PaymentSchedules []PaymentSchedule `json:"payment_schedules,omitempty"`
+}
+
+type PaymentSchedule struct {
+	Amount                *decimal.Decimal `json:"amount,omitempty"`
+	Currency              string           `json:"currency,omitempty"`
+	IssuedAt              string           `json:"issued_at,omitempty"`
+	DueAt                 string           `json:"due_at,omitempty"`
+	CompletedAt           string           `json:"completed_at,omitempty"`
```
Comment (oliver006) [inline order.go:303]: Is this a date? If yes, do you need a different type? Also, does this need to be `*` as it can be null?

File: order.go [hunk 3]
```diff
@@ -285,6 +286,24 @@ type RefundLineItem struct {
...
+	PaymentTermsType string            `json:"payment_terms_type,omitempty"`
+	DueInDays        string            `json:"due_in_days,omitempty"`
+	PaymentSchedules []PaymentSchedule `json:"payment_schedules,omitempty"`
+}
+
+type PaymentSchedule struct {
+	Amount                *decimal.Decimal `json:"amount,omitempty"`
+	Currency              string           `json:"currency,omitempty"`
+	IssuedAt              string           `json:"issued_at,omitempty"`
+	DueAt                 string           `json:"due_at,omitempty"`
```
Comment (oliver006) [inline order.go:302]: This looks to be a date?

File: order.go (hunk 2 above)
Comment (oliver006) [inline order.go:303]: Same

**Review by oliver006 — COMMENTED** [review]
Left a couple of comments.
Can you also add some payment terms to a fixture and add a test? In particular around the dates and that they work when set to null? Prob best to check the docs when this can be the case.
